pinc/misc.inc

```diff
@@ -732,6 +732,30 @@ function extract_zip_to(string $path_to_zip, string $output_directory): bool
     // No need to check the result of the function since it is checked in is_valid_zip_file
     $zip->open($path_to_zip);
 
+    if($remove_top_dir)
+    {
+        // find if there is a common prefix implying a common directory, remove it
+        $filenames = [];
+        $last_index = $zip->numFiles - 1;
+        for ($i = 0; $i <= $last_index; $i++)
+        {
+            $filenames[] = $zip->getNameIndex($i);
+        }
+        sort($filenames);
+        $prefix = $filenames[0];
+        if(endswith($prefix, "/") && startswith($filenames[$last_index], $prefix))
```

Comment:
Adding a comment above this line on what it is doing would help further devs. something like:
```
// If the first entry is a directory (ending in /) and the last entry
// starts with that directory, then all files in the archives are within
// the directory and we remove it from all files in the archive.
```